python/ray/serve/_private/autoscaling_state.py

```diff
@@ -270,15 +312,19 @@ def get_decision_num_replicas(
         `_skip_bound_check` is True, then the bounds are not applied.
         """
 
-        decision_num_replicas = self._policy(
-            curr_target_num_replicas=curr_target_num_replicas,
-            total_num_requests=self.get_total_num_requests(),
-            num_running_replicas=len(self._running_replicas),
-            config=self._config,
-            capacity_adjusted_min_replicas=self.get_num_replicas_lower_bound(),
-            capacity_adjusted_max_replicas=self.get_num_replicas_upper_bound(),
-            policy_state=self._policy_state,
+        self._autoscaling_context.target_num_replicas = curr_target_num_replicas
+        self._autoscaling_context.total_num_requests = self.get_total_num_requests()
+        self._autoscaling_context.running_replicas = len(self._running_replicas)
+        self._autoscaling_context.config = self._config
+        self._autoscaling_context.capacity_adjusted_min_replicas = (
+            self.get_num_replicas_lower_bound(),
+        )
+        self._autoscaling_context.capacity_adjusted_max_replicas = (
+            self.get_num_replicas_upper_bound(),
         )
+        self._autoscaling_context.policy_state = self._policy_state
```

Comment:
let's reconstruct the `_autoscaling_context` object instead of mutating one inplace